Detectors/TPC/reconstruction/include/TPCReconstruction/DigitalCurrentClusterIntegrator.h

```diff
@@ -0,0 +1,55 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file TPCCATracking.h
+/// \brief Decoder to convert TPC ClusterHardware to ClusterNative
+/// \author David Rohr
+#ifndef ALICEO2_TPC_DIGITALCURRENTCLUSTERINTEGRATOR_H_
+#define ALICEO2_TPC_DIGITALCURRENTCLUSTERINTEGRATOR_H_
+
+#include <vector>
+#include <memory>
+
+#include "TPCReconstruction/ClusterNative.h"
+#include "TPCBase/Constants.h"
+#include "TPCBase/Mapper.h"
+
+namespace o2 { namespace TPC {
+
+//This class contains an array or all TPC pads (in selected sectors and rows), and can integrated the charges of clusters
+class DigitalCurrentClusterIntegrator
+{
+public:
+  DigitalCurrentClusterIntegrator() = default;
+  ~DigitalCurrentClusterIntegrator() = default;
+  
+  void initRow(int sector, int row) {
+    if (mIntegratedCurrents[sector][row] == nullptr) {
+      int nPads = o2::TPC::Mapper::instance().getNumberOfPadsInRowSector(row);
+      mIntegratedCurrents[sector][row].reset(new unsigned long long int[nPads]);
+      memset(&mIntegratedCurrents[sector][row][0], 0, nPads * sizeof(mIntegratedCurrents[sector][row][0]));
+    }
+  }
+  void integrateCluster(int sector, int row, float pad, unsigned int charge) {
+    int ipad = ipad + 0.5;
+    if (ipad < 0) ipad = 0;
+    int maxPad = o2::TPC::Mapper::instance().getNumberOfPadsInRowSector(row);
+    if (ipad >= maxPad) ipad = maxPad - 1;
+    mIntegratedCurrents[sector][row][ipad] += charge;
+  }
```

Comment:
So, if I understand correctly, this integrates on a per-pad level. At the end we need some fiducial volume, which still needs to be defined, but could be on the order of 5 rows x 5 pads x 1ms, or even large in pad and row direction.

I guess this will be an extension in future.

Reply:
I agree, but as long as this is not defined, this code at least somehow simulates memory and CPU load, as requested by Gvozden.

Reply:
More or less. At the end we will need robust estimators in order to remove e.g. very noisy pads or pads in saturation. This might require sorting and more.

But it is definitely a good start.